src/inputs/pipe.c

```diff
@@ -302,6 +305,42 @@ parse_progress(struct input_metadata *m, char *progress)
   m->song_length = (end - start) * 10 / 441; // Convert to ms based on 44100
 }
 
+static void
+parse_volume(char *volume)
+{
+  char *value_begin;
+  char *value_end;
+  float airplay_volume;
+
+  if (!(value_begin = strtok_r(volume, ",", &value_end)))
+    return;
+
+  airplay_volume = strtof(value_begin, &value_end);
+
+  if ((errno == ERANGE) || (errno != 0 && airplay_volume == 0) || (value_begin == value_end))
+    {
+      DPRINTF(E_LOG, L_PLAYER, "Invalid airplay volume in string (%s): %s\n", volume, strerror(errno));
+      return;
+    }
+
+  if (fabs(airplay_volume + 144) < 1)
+    {
+      DPRINTF(E_DBG, L_PLAYER, "Applying Shairport/Airplay volume 'mute' (value: %.2f)\n", airplay_volume);
+      player_volume_set(0);
+    }
+  else if (airplay_volume >= -30 && airplay_volume <= 0)
+    {
+      float local_volume = roundf((float)100 + (airplay_volume / (float)30 * (float)100));
```

Comment:
Your right my bad, this is absolutely not necessary, should think more before typing ;-)